src/mappings/eventHandlers.ts

```diff
@@ -74,12 +77,19 @@ async function findOrCreateBlock(substrateBlock: SubstrateBlock): Promise<String
     return existingBaseBlock.id;
   }
 
+  // Calculate weight
+  let blockWeight = new BN(0);
+  events.forEach((event) => {
+    const { event: { method, section, data } } = event;
+    const weight = section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method) ? ((method === 'ExtrinsicSuccess' ? data[0] : data[1]) as DispatchInfo).weight : new BN(0);
```

Comment:
would this calculation be accurate? it seems like you're only doing weight calculations off of events emitted from extrinsics. There may be many things that happen where the chain doesn't really emit events. So just wanted to check to make sure whether this captures the true weight of the block. I feel like at least all of the block overhead wouldn't be here, as well as all any extrinsics/actions on the block that don't emit an event. unless i'm misunderstanding what an event is (my understanding: see picture below)

![image](https://user-images.githubusercontent.com/22332240/178010070-aa782614-50cb-4aab-9742-ce8c649cce3e.png)